gdal/frmts/netcdf/netcdfsg.cpp

```diff
@@ -0,0 +1,1047 @@
+		}
+	
+
+		// (3) check touple order
+		if(this->touple_order < 2)
+		{
+			throw new SG_Exception_Existential(container_name, "insufficent node coordinates must have at least two axis");	
+		}
+
+		this->pt_buffer = new Point(this->touple_order);
+		
+		// Set other values accordingly
+		this->gc_varId = geoVarId; 
+		this->current_vert_ind = 0;	
+		this->ncid = ncId;
+	}
+
+	SGeometry::~SGeometry()
+	{
+		delete this->pt_buffer;
+	}
+
+	Point& SGeometry::next_pt()
+	{
+		if(!this->has_next_pt())
+		{
+			throw new SG_Exception_BadPoint();
+		}
+
+		// Fill pt
+		// New pt now
+		for(int order = 0; order < touple_order; order++)
+		{
+			Point& pt = *(this->pt_buffer);
+			double data;
+			size_t full_ind = bound_list[cur_geometry_ind] + current_vert_ind;
+
+			// Read a single coord
+			int err = nc_get_var1_double(ncid, nodec_varIds[order], &full_ind, &data);
+			// To do: optimize through multiple reads at once, instead of one datum
+
+			if(err != NC_NOERR)
+			{
+				throw new SG_Exception_BadPoint();
+			}
+
+			pt[order] = data;
+		}	
+		
+		this->current_vert_ind++;
+		return *(this->pt_buffer);	
+	}
+
+	bool SGeometry::has_next_pt()
+	{
+		if(this->current_vert_ind < node_counts[cur_geometry_ind])
+		{
+			return true;
+		}
+	
+		else return false;
+	}
+
+	void SGeometry::next_geometry()
+	{
+		// to do: maybe implement except. and such on error conds.
+
+		this->cur_geometry_ind++;
+		this->cur_part_ind = 0;
+		this->current_vert_ind = 0;	
+	}
+
+	bool SGeometry::has_next_geometry()
+	{
+		if(this->cur_geometry_ind < node_counts.size())
+		{
+			return true;
+		}
+		else return false;
+	}
+
+	Point& SGeometry::operator[](int index)
+	{
+		for(int order = 0; order < touple_order; order++)
+		{
+			Point& pt = *(this->pt_buffer);
+			double data;
+			size_t real_ind = index;
+
+			// Read a single coord
+			int err = nc_get_var1_double(ncid, nodec_varIds[order], &real_ind, &data);
+
+			if(err != NC_NOERR)
+			{
+				throw new SG_Exception_BadPoint();
+			}
+
+			pt[order] = data;
+		}	
+
+		return *(this->pt_buffer);
+	}
+
+	size_t SGeometry::get_geometry_count()
+	{
+		if(type == POINT)
+		{
+			// If nodes global attribute is available, use that
+
+			// Otherwise, don't fail- use dimension length of one of x
+
+			if(this->nodec_varIds.size() < 1) return 0;
+
+			// If more than one dim, then error. Otherwise inquire its length and return that
+			int dims;
+			if(nc_inq_varndims(this->ncid, nodec_varIds[0], &dims) != NC_NOERR) return 0;
+			if(dims != 1) return 0;
+			
+			// Find which dimension is used for x
+			int index;
+			if(nc_inq_vardimid(this->ncid, nodec_varIds[0], &index) != NC_NOERR)
+			{
+				return 0;
+			}
+
+			// Finally find the length
+			size_t len;
+			if(nc_inq_dimlen(this->ncid, index, &len) != NC_NOERR)
+			{
+				return 0;
+			}
+			return len;	
+		}
+
+		else return this->node_counts.size();
+	}
+
+	/* serializeToWKB(SGeometry * sg)
```

Comment:
high level question: why doing hand-written wkb serialization and not using OGRGeometry and derived classes methods directly. Would be more straightforward

Reply:
ooh well see you guys have a very intricate class hierarchy and thought that doing this type of byte by byte low level serialization at the time would be easier to do navigate then go through each specific part of the class hierarchy, though to be honest I usually go low level if it's not a big burden ;)